internal/daemon/worker/session/session.go

```diff
@@ -214,9 +214,13 @@ func (s *sess) GetLocalConnections() map[string]ConnInfo {
 	return res
 }
 
+// Return local tofu token, if availble; otherwise return the tofu token from the resp
 func (s *sess) GetTofuToken() string {
 	s.lock.RLock()
 	defer s.lock.RUnlock()
+	if len(s.tofuToken) > 0 {
+		return s.tofuToken
+	}
```

Comment:
Can you also add a nil check on Resp, in case it's nil? I"m not sure why it would be, but seems like a good idea...